src/amqp_management.c

```diff
@@ -281,14 +279,53 @@ static void on_message_sender_state_changed(void* context, MESSAGE_SENDER_STATE
         break;
 
     case MESSAGE_SENDER_STATE_OPEN:
-        amqp_management_instance->sender_connected = 1;
-        (void)send_operation_messages(amqp_management_instance);
+        switch (amqp_management_instance->amqp_management_state)
+        {
+        default:
+        case AMQP_MANAGEMENT_STATE_ERROR:
+            /* do nothing */
```

Comment:
I am not re-writing with tests amqp_management.c as part of this change. This will come in a subsequent PR in order to keep things scoped.